src/Kiota.Builder/KiotaBuilder.cs

```diff
@@ -1849,11 +1849,21 @@ internal static void AddDiscriminatorMethod(CodeClass newClass, string discrimin
     }
     private void CreatePropertiesForModelClass(OpenApiUrlTreeNode currentNode, OpenApiSchema schema, CodeNamespace ns, CodeClass model)
     {
-        if (!model.Properties.Any(static x => x.IsOfKind(CodePropertyKind.Custom))) // this redundant check on model properties is here to avoid race conditions
-            if (schema?.Properties?.Any() ?? false)
+        if (!model.IsPropertiesBuilt())
+        {
+            try
             {
-                model.AddProperty(schema
-                                    .Properties
+                model.StartBuildingProperties();
```

Comment:
it does feel odd the builder has to manage the internal state of the object in additions to building it. Could we have the model already be "started" at least?

Reply:
I'm not sure if that would work. Starting building enters a monitor. I could try to externalize the building states from CodeClass.

Reply:
I've tried creating the new lifecycles 'started', but the problem is that it's not guaranteed that the thread creating the lifecycle will also be the one building the properties.